openedx/core/djangoapps/user_api/preferences/tests/test_views.py

```diff
@@ -288,6 +292,56 @@ def test_update_preferences_other_user(self, api_client, user):
         )
 
 
+@unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+class TestPreferencesAPITransactions(TransactionTestCase):
+    """
+    Tests the transactional behavior of the preferences API
+    """
+    test_password = "test"
+
+    def setUp(self):
+        super(TestPreferencesAPITransactions, self).setUp()
+        self.client = APIClient()
+        self.user = UserFactory.create(password=self.test_password)
+        self.url = reverse("preferences_api", kwargs={'username': self.user.username})
+
+    @patch('openedx.core.djangoapps.user_api.models.UserPreference.delete')
+    def test_update_preferences_rollback(self, delete_serializer):
+        """
+        Verify that updating preferences is transactional when a failure happens.
+        """
+        # Create some test preferences values.
+        set_user_preference(self.user, "a", "1")
+        set_user_preference(self.user, "b", "2")
+        set_user_preference(self.user, "c", "3")
+
+        # Patch the serializer so that it throws an exception on the second save
+        # serializer_save = patch('openedx.core.djangoapps.user_api.serializers.RawUserPreferenceSerializer.save')
```

Comment:
That was my failed attempt to have one of the updates patched but not another. I eventually settled on patching delete instead which has the same effect. I've removed the bogus comment (and fixed the name of the patch variable).